File: Engine/src/com/shinkson47/opex/backend/runtime/threading/OPEXGame.java
package com.shinkson47.opex.backend.runtime.threading;

import com.shinkson47.opex.backend.toolbox.Versionable;

/**
 * Abstract for user's class. Indicates class as a valid game main for
 * OPEX.
 *
 * @author Jordan Gray
 * @version 2
 * @since 2020.7.11.A
 */
public abstract class OPEXGame extends Versionable implements IOPEXRunnable {}

File: Engine/src/com/shinkson47/opex/backend/runtime/threading/OPEXGame.kt
package com.shinkson47.opex.backend.runtime.threading

import com.shinkson47.opex.backend.runtime.IOPEXVersionable

/**
 * Super interface for user's class. Indicates class as a valid game main for
 * OPEX.
 *
 * @author gordie
 */
interface OPEXGame : IOPEXRunnable, IOPEXVersionable 